src/instana/instrumentation/aws/boto3.py

```diff
@@ -0,0 +1,143 @@
+# (c) Copyright IBM Corp. 2025
+
+from typing import TYPE_CHECKING, Any, Callable, Dict, Sequence, Tuple, Type
+
+from instana.instrumentation.aws.dynamodb import collect_dynamodb_attributes
+from instana.instrumentation.aws.s3 import collect_s3_attributes
+from opentelemetry.semconv.trace import SpanAttributes
+
+if TYPE_CHECKING:
+    from botocore.auth import SigV4Auth
+    from botocore.client import BaseClient
+
+    from instana.span.span import InstanaSpan
+
+try:
+    import json
+
+    import wrapt
+
+    from instana.log import logger
+    from instana.propagators.format import Format
+    from instana.singletons import tracer
+    from instana.span.span import get_current_span
+    from instana.util.traceutils import (
+        extract_custom_headers,
+        get_tracer_tuple,
+        tracing_is_off,
+    )
+
+    def lambda_inject_context(payload: Dict[str, Any], span: "InstanaSpan") -> None:
+        """
+        When boto3 lambda client 'Invoke' is called, we want to inject the tracing context.
+        boto3/botocore has specific requirements:
+        https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/lambda.html#Lambda.Client.invoke
+        """
+        try:
+            invoke_payload = payload.get("Payload", {})
+
+            if not isinstance(invoke_payload, dict):
+                invoke_payload = json.loads(invoke_payload)
+
+            tracer.inject(span.context, Format.HTTP_HEADERS, invoke_payload)
+            payload["Payload"] = json.dumps(invoke_payload)
+        except Exception:
+            logger.debug("non-fatal lambda_inject_context: ", exc_info=True)
+
+    @wrapt.patch_function_wrapper("botocore.auth", "SigV4Auth.add_auth")
+    def emit_add_auth_with_instana(
+        wrapped: Callable[..., None],
+        instance: "SigV4Auth",
+        args: Tuple[object],
+        kwargs: Dict[str, Any],
+    ) -> Callable[..., None]:
+        current_span = get_current_span()
+        if not tracing_is_off() and current_span and current_span.is_recording():
+            extract_custom_headers(current_span, args[0].headers)
+        return wrapped(*args, **kwargs)
+
+    @wrapt.patch_function_wrapper("botocore.client", "BaseClient._make_api_call")
+    def make_api_call_with_instana(
+        wrapped: Callable[..., Dict[str, Any]],
+        instance: Type["BaseClient"],
+        args: Sequence[Dict[str, Any]],
+        kwargs: Dict[str, Any],
+    ) -> Dict[str, Any]:
+        # If we're not tracing, just return
+        if tracing_is_off():
+            return wrapped(*args, **kwargs)
+
+        tracer, parent_span, _ = get_tracer_tuple()
+
+        parent_context = parent_span.get_span_context() if parent_span else None
+
+        try:
+            if instance:
```

Comment:
Is this `if` really necessary?

Reply:
Oh I forgot to remove it after modifying the code for lamdba and s3